app/api/routes.py

```diff
@@ -179,6 +182,57 @@ def get_resources():
     return standardize_response(payload=dict(data=resource_list, **pagination_details))
 
 
+def search(term):
+    query = Resource.query
+    page = 0
+
+    if request.args.get('page') is not None:
+        page = int(request.args.get('page'))
+
+    indicies = client.list_indices()
+    for ind in indicies['items']:
+        if ind['name'] == 'resources_api':
+            curr_index = ind
+
+            if curr_index['entries'] != query.count():
+                print("ReIndexing Database")
+
+                index.clear_objects()
+                db_list = [u.serialize for u in query.all()]
+                index.save_objects(db_list)
+
+            break
+
+    search_results = index.search(f'{term}', {
+        'attributesToRetrieve': [
+            'name',
+            'url',
+            'category',
+            'languages',
+            'notes',
+        ],
+        'page': page,
+    })
+
+    if page >= int(search_results['nbPages']):
+        return redirect('/404')
+
+    # Translate search to normal format.
+    results = []
+    for result in search_results['hits']:
+        results.append(query.get(int(result['objectID'])).serialize)
+
+    pagination_details = {
+                "pagination_details": {
+                    "page": search_results['page'],
+                    "number_of_pages": search_results['nbPages'],
+                    "records_per_page": search_results['hitsPerPage'],
+                    "total_count": search_results['nbHits'],
+                }
+        }
+    return standardize_response(payload=dict(data=results, **pagination_details))
```

Comment:
Since Algolia does its own pagnation, we dont need to use the pagnator to 
1. Pagnate the data
2. Generate the pagnation details.

There's probably a better way to handle this.

Reply:
Also, our paginator starts at 1 and Algolias starts at 0. I dont know if that is something we should care about

Reply:
Indexes should start at 0

Reply:
You know, you're right. Our pagination does start at 1, I never realized that was weird.

Did a bit of digging, and it turns out that this was a choice by the makers of Flask-SQLAlchemy, which we use as our ORM https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/#flask_sqlalchemy.Pagination.page

I'm not sure why they chose to make it 1 indexed, but translating that to a 0 index would certainly be more work than its worth, even though Kyle is right that it probably ought to be 0 indexed